### PR TITLE
Regression in mod_menu - base item type other than component not showing

### PR DESCRIPTION
**Pull request by Devportobello**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/15617 .

### Summary of Changes

Added 1 parameter to XML modal_menu field to allow possibility to enable type of menu item

### Testing Instructions

Create 1 menu eg: Main
Add to this menu some items like this:

Level - Item - Type
1 - Home - Article
2 - Activities - Menu heading
2.1 - Activity 1 - Article
2.2 - Activity 2 - Article
2.3 - Activity 3 - Article
3 - Contact - Article
image

Then go to modules > add a new module "Menu"
Select Menu : Main
Select base item : Activities
image

### Expected result

Could select "Activites"

### Actual result

Unable to see and select "Activities"

### Final point

Maybe could better to use a disable parameter, if right, we need to modify all modal_menu to B/C

### CODE REVIEW

**Comment by infograf768** [comment]
Was that working in 3.6.5?

**Comment by Devportobello** [comment]
Yes working in 3.6.5; prior of https://github.com/joomla/joomla-cms/commit/33d7bd916df6936055109504c0ac1cc5df260787

**Comment by aaron-harding** [comment]
Just applied this to a test site. When opening the modal and applying a filter, the whole menu item list is empty. The form submit on the filter is removing the "enable" param.

**Comment by Devportobello** [comment]
local and fork repo was in bad situation... how can i erase this last commits T_T

**Comment by crofton77** [comment]
Any update on getting this in to Joomla release? It really annoying I can't set the base item for menus when they are aliases

**Comment by infograf768** [comment]
To get this in, this PR has first to be updated to solve the conflicts.

**Comment by brianteeman** [comment]
and it has to have successful tests

**Comment by infograf768** [comment]
> and it has to have successful tests

right and at first sight I am not sure it is B/C as some other `modal_menu` fields do not have a `disable` attrib (some redirects where one could choose a Heading as redirect) or creating an alias menu item aliased to another alias menu item, etc.)
But we really can't test until updated.

**Comment by Devportobello** [comment]
This need serious test, too much case...

**Comment by SharkyKZ** [comment]
@Devportobello Are you still around? Can you fix the conflicts?

File: administrator/components/com_menus/views/items/tmpl/modal.php [hunk 1]
```diff
@@ -86,7 +88,7 @@
 				</tfoot>
 				<tbody>
 				<?php foreach ($this->items as $i => $item) : ?>
-				<?php $uselessMenuItem = in_array($item->type, array('separator', 'heading', 'alias', 'url', 'container')); ?>
+					<?php if ( !in_array($item->type, $enable) ) continue; ?>
```
Comment (izharaazmi) [inline administrator/components/com_menus/views/items/tmpl/modal.php:91]: I think we should not just skip that record altogether. We should show that record with click action disabled, as in previous.

In short just replace with 

```php
$uselessMenuItem = !in_array($item->type, $enable);
```

rest all can remain as is.

Comment (izharaazmi) [reply administrator/components/com_menus/views/items/tmpl/modal.php:91]: Also the variable name `$uselessMenuItem` may be renamed to something like `$disabled`

**Comment by zero-24** [comment]
Dear @Devportobello 

in preperation of the upcomming release of Joomla 3.10 we have used GitHubs rename feature to rename the staging branch into 3.10-dev. Usually GitHub moves all existing PRs towards the new branch just fine, but here it didnt work. The reason seems to be that the fork of the CMS that was used as base for this PR has been deleted so GitHub does no longer have a base to rebase the PR against the new branch and we are also not able to reopen the PR. For that reason GitHub closed this PR in my name, when this issue is still valid It would require a new PR against the new 3.10-dev or 4.0-dev branch.

**Pull request closed without merging**